pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java

```diff
@@ -89,12 +96,16 @@ public static QueryScheduler create(PinotConfiguration schedulerConfig, QueryExe
 
   @Nullable
   private static QueryScheduler getQuerySchedulerByClassName(String className, PinotConfiguration schedulerConfig,
-      QueryExecutor queryExecutor) {
+      QueryExecutor queryExecutor, ServerMetrics serverMetrics, LongAccumulator latestQueryTime) {
     try {
-      Constructor<?> constructor =
-          Class.forName(className).getDeclaredConstructor(Configuration.class, QueryExecutor.class);
+      Constructor<?> constructor = Class.forName(className)
+          .getDeclaredConstructor(PinotConfiguration.class, QueryExecutor.class, ServerMetrics.class,
+              LongAccumulator.class);
       constructor.setAccessible(true);
-      return (QueryScheduler) constructor.newInstance(schedulerConfig, queryExecutor);
+      QueryScheduler queryScheduler =
+          (QueryScheduler) constructor.newInstance(schedulerConfig, queryExecutor, serverMetrics, latestQueryTime);
+      LOGGER.info("Using {} scheduler", queryScheduler.name());
```

Comment:
maybe move this log statement to line 81 (and perhaps use a single log statement at the end of the switch there?)

Reply:
Done